### DISCUSSION TITLE
UI Changes Shop Backend

### DISCUSSION THREAD

**Discussion by babsgosgens** [gameserverapp/Platform#929] [Suggestions]
I am proposing several, related features to the Shop backend:

# 
**Publish/Unpublish attributes**
Unpublished shop packs will never be included in the frontends. This allows for packs to be fully prepared ahead of time (in bulk) without having to worry about changing the visibility afterward.

Unpublished behaves the same as the `Hidden for all` attribute.
Published however, is not the same as `Visible to anyone` feature. A pack that is `Visible to all` may not be meant to be online just yet. Having it unpublished, will allow for easier batch editing (see feature proposed below). New packs can be prepared ahead of time (set to be visible to all) but will only become available as soon as the author *publishes* them.

I see this working alongside a cloning and batch update feature; clones will be unpublished by default and can be batch selected to be published when ready.

# 
**Batch edit in backend**
Add a checkbox to select multiple shoppacks apply one of these actions:

- Publish
- Unpublish
- Clone (always unpublished)
- Delete

![image](https://github.com/user-attachments/assets/90538b0e-b287-4dad-8f4e-2d69d4bdd79e)

# 
**Add Save as feature to Shop pack view**
This is essentially a cloning feature. When using “Save as..”, an unpublished clone of the pack will be created and opened ready for editing in a (new) view.

![image](https://github.com/user-attachments/assets/8b01ed56-7b0d-4a0b-b6c4-f7b26ba197e4)

**Comment by aw08-2004** [comment]
Also Add Bulk Actions
 -- Delete Multiple
 -- Add/Remove/Move Group
